autocomplete/complete.go

```diff
@@ -217,7 +217,15 @@ func getPotentialPaths(ctx context.Context, resolver *buildcontext.Resolver, gwC
 func getPotentialBuildArgs(ctx context.Context, resolver *buildcontext.Resolver, gwClient gwclient.Client, targetStr string) ([]string, error) {
 	target, err := domain.ParseTarget(targetStr)
 	if err != nil {
-		return nil, err
+		// An artifact path ("earthly -a +target/out -<TAB>") will land us here as the separator fails
```

Comment:
What about renaming the original `getPotentialBuildArgs` to `getPotentialBuildArgsForTarget` and creating a new function called `getPotentialBuildArgsForArtifact`  which calls `domain.ParseArtifact(targetStr)` instead.

This would get around the case of doing autocompletion for an invalid command such as `earthly +othertargetwithargs/out`.

then as we're parsing the flags, if we see the `--artifact` flag has been specified, then we'll know to call `getPotentialBuildArgsForTarget` instead.